packages/node_modules/@ciscospark/react-redux-spark/src/actions.js

```diff
@@ -21,8 +23,15 @@ export function storeSparkInstance(spark) {
 
 export function registerDevice(spark) {
   return (dispatch) => {
+    metrics.saveElapsedTime(`registerDevice:start`);
+    metrics.sendTempMetric(`registerDevice:start`, spark);
     dispatch(updateSparkStatus({registering: true}));
     return spark.device.register()
-      .then(() => dispatch(updateSparkStatus({registering: false, registered: true})));
+      .then(() => {
+        metrics.saveElapsedTime(`registerDevice:end`);
+        metrics.sendTempMetric(`registerDevice:end`, spark);
+        metrics.sendDurationMetric(`registerDevice:duration`, `registerDevice:start`, `registerDevice:end`, spark);
+        dispatch(updateSparkStatus({registering: false, registered: true}));
```

Comment:
We could `return dispatch` here to keep it thunkable (?), but I don't think it is needed.